src/sentry/integrations/gitlab/search.py

```diff
@@ -1,82 +1,64 @@
-from typing import Any
-
-from rest_framework.request import Request
 from rest_framework.response import Response
 
-from sentry.api.api_owners import ApiOwner
-from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import control_silo_endpoint
-from sentry.integrations.api.bases.integration import IntegrationEndpoint
 from sentry.integrations.gitlab.integration import GitlabIntegration
 from sentry.integrations.models.integration import Integration
-from sentry.organizations.services.organization import RpcOrganization
+from sentry.integrations.source_code_management.search import SourceCodeSearchEndpoint
 from sentry.shared_integrations.exceptions import ApiError
 
 
 @control_silo_endpoint
-class GitlabIssueSearchEndpoint(IntegrationEndpoint):
-    owner = ApiOwner.INTEGRATIONS
-    publish_status = {
-        "GET": ApiPublishStatus.PRIVATE,
-    }
-
-    def get(
-        self, request: Request, organization: RpcOrganization, integration_id: int, **kwds: Any
-    ) -> Response:
-        try:
-            integration = Integration.objects.get(
-                organizationintegration__organization_id=organization.id,
-                id=integration_id,
-                provider="gitlab",
-            )
-        except Integration.DoesNotExist:
-            return Response(status=404)
+class GitlabIssueSearchEndpoint(SourceCodeSearchEndpoint):
+    @property
+    def repository_field(self):
+        return "project"
 
-        field = request.GET.get("field")
-        query = request.GET.get("query")
-        if field is None:
-            return Response({"detail": "field is a required parameter"}, status=400)
-        if query is None:
-            return Response({"detail": "query is a required parameter"}, status=400)
+    @property
+    def integration_provider(self):
+        return "gitlab"
 
-        installation = integration.get_installation(organization.id)
-        assert isinstance(installation, GitlabIntegration), installation
+    @property
+    def installation_class(self):
+        return GitlabIntegration
 
-        if field == "externalIssue":
-            project = request.GET.get("project")
-            if project is None:
-                return Response({"detail": "project is a required parameter"}, status=400)
-            try:
-                iids = [int(query)]
-                query = None
-            except ValueError:
-                iids = None
+    def handle_search_issues(self, installation, query: str, repo: str) -> Response:
```

Comment:
for example, installation here would ideally be GitlabIntegration (or at least IssueBasicIntegration) so that the IDE can recognize the return type of `installation.search_issues`...